File: browser/src/rpc/graph.js
import rpc from './vertex_pb';
import grpcFetch from './grpcfetch';


export default function () {
  let request = new rpc.QueryRequest({});
  request.setText("hi");

  grpcFetch('/rpc.Graph/Query', request.serializeBinary()).then((bytes) => {
    var message2 = rpc.QueryReply.deserializeBinary(bytes);
    console.log('message2', message2.getText());
    console.log('message2', message2.getPropertiesMap().get('test').getId());
  })
};

File: rpc/graph.go
package rpc

import (
	"github.com/Sirupsen/logrus"
	context "golang.org/x/net/context"
)

type Graph struct {
}

func (g Graph) Query(context.Context, *QueryRequest) (*QueryReply, error) {
	logrus.Print("hit")
	v := &Vertex{Id: "test"}
	m := make(map[string]*Vertex)
	m[v.GetId()] = v
	reply := &QueryReply{Properties: m, Text: "hi ross"}
	return reply, nil
}
